internal/extproc/processor.go

```diff
@@ -212,6 +220,19 @@ func (p *Processor) maybeBuildDynamicMetadata() (*structpb.Struct, error) {
 			cost = p.costs.OutputTokens
 		case filterconfig.LLMRequestCostTypeTotalToken:
 			cost = p.costs.TotalTokens
+		case filterconfig.LLMRequestCostTypeCELExpression:
+			costU64, err := llmcostcel.EvaluateProgram(
+				c.celProg,
+				p.requestHeaders[p.config.modelNameHeaderKey],
+				p.requestHeaders[p.config.selectedBackendHeaderKey],
+				p.costs.InputTokens,
+				p.costs.OutputTokens,
+				p.costs.TotalTokens,
+			)
+			if err != nil {
+				return nil, fmt.Errorf("failed to evaluate CEL expression: %w", err)
```

Comment:
Out of curiosity, what does it mean for ext_proc to fail at this stage, when the response headers may have been sent? Does it make sense to make it fail?

Reply:
this will result in connection termination i guess